docs/addons/writing-addons.md

```diff
@@ -0,0 +1,226 @@
+| src               | Source code for the addon         |
+| .babelrc.js       | Babel configuration               |
+| register.js       | Addon entry point                 |
+| package.json      | Addon metadata information        |
+| README.md         | General information for the addon |
+
+### Get started
+
+Open a new terminal and create a new directory called `my-addon`. Inside it run `npm init` to initialize a new node project. For project's name enter `my-addon` and for entry point choose `dist/register.js`. 
+
+Once you've gone through the prompts your `package.json` should look like so:
+
+```json
+{
+  "name": "my-addon",
+  "version": "1.0.0",
+  "description": "A barebones Storybook addon",
+  "main": "dist/register.js",
+  "files": [
+    "dist/**/*",
+    "README.md",
+    "*.js"
+  ],
+  "keywords": [
+    "storybook",
+    "addons"
+  ],
+  "author": "YourUsername",
+  "license": "MIT"
+}
+```
+
+### Build system
+
+We'll need to add the necessary dependencies and make some adjustments. With a terminal opened inside your addon directory issue the following commands:
+
+```shell
+# Installs React
+yarn add react react-dom
+
+# Adds Storybook:
+npx -p @storybook/cli sb init
+```
+
+<div class="aside">
+Initializing a new Storybook installation through <code>npx -p @storybook/cli sb init</code> will grant us access to the necessary Storybook packages for our addon. If you're 
+building your own standalone Storybook addon you'll need to set both React and Storybook packages as peer dependencies. Applying this change will prevent the addon from breaking Storybook in case there's different package versions available.
+</div>
+
+Next create a `.babelrc.js` file in the root directory with the following:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/my-addon-babel-configuration.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+Because the addon we're about to write will use ES6 and rely on JSX, thus we'll need to include Babel and configure it with the proper presets and options. 
+</div>
+
+Change your `package.json` and add the following script to build the addon:
+
+```json
+{
+  "scripts": {
+    "build": "babel ./src --out-dir ./dist"
+  },
+}
+```
+<div class="aside">
+Running <code>yarn build</code> at this stage will output the code into the <code>dist</code> directory, transpiled into a ES5 module ready to be installed into any Storybook. 
+</div>
+
+### Add a panel
+
+Now let’s add a panel to Storybook. Create a new directory called `src` and inside a new file called `register.js`.
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-addon-panel-initial.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+This is boilerplate code for any addon that adds a panel to Storybook, and there’s really not much going on here. In this case, we’re just adding a static `div` that renders when the panel is selected in Storybook’s UI.
```

Comment:
I'm going to argue there is quite a bit happening here. We could say it's boilerplate. But, let's explain what exactly is happening. i.e., `register`, `add` and the `key` bit which is super important and without it you get errors.

Reply:
Updated with the latest commit.